.ci/ci-fast-return.sh

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Description: Perform CI 'fast path return' checks.
+# Returns: 1 for 'fast return OK', 0 for 'do not fastpath'.
+#
+# Checks for:
+#  - file name patterns from the YAML that will force the CI to run
+#  - file name patterns from the YAML to see if we can skip all files in the PR
+#  - If potentially skipping, looks for a 'force' label on the PR to force the CI
+#    to run anyway
+
+set -e
+
+[ -n "$DEBUG" ] && set -x
+
+cidir=$(dirname "$0")
+source "${cidir}/lib.sh"
+
+# If no branch specified, compare against the master.
+# The 'branch' var is required by the get_pr() lib functions.
+branch=${branch:-master}
+
+# The YAML file containing our filename match patterns.
+yqfile="${cidir}/ci-fast-return.yaml"
+
+# Name of the label, that if set on a PR, will force the CI to be run anyway.
+force_label="force-ci"
+
+# Check if any files changed by the PR either force the CI to run or if
+# we can skip all the files in the PR.
+#
+# Return:
+#  0 - all files are fastpath
+#  1 - at least one non-fastpath file found
+can_we_skip() {
+	# The branch is the baseline - ignore it.
+	[ "$specific_branch" = "true" ] && info "Skip basline branch" >&2 && echo "0" && return
```

Comment:
Typo: "baseline". Although I do like "one-liners" like this, I think due to all the ampersands this would be clearer as a full `if ... then` test. Also, since this is a `set -e` script, it's always a good idea to return an explicit value as a "bare `return`" can be surprising since it will return the value of the last command executed.